Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -100,9 +112,92 @@ class HistoryPanel: SiteTableViewController, HomePanel {
         }
     }
 
+    func updateNumberOfSyncedDevices() {
+        var numberOfDevices = 0
+        syncedTabsHeaderCell.descriptionLabel?.hidden = true
+        syncedTabsHeaderCell.descriptionLabel?.removeFromSuperview()
+        syncedTabsHeaderCell.descriptionLabel = nil
+
+        self.profile.getCachedClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
+            if let clientAndTabs = result.successValue {
+                let nonEmptyClientAndTabs = clientAndTabs.filter { $0.tabs.count > 0 }
+                numberOfDevices = nonEmptyClientAndTabs.count
+            }
+
+            if numberOfDevices > 0 {
+                self.syncedTabsHeaderCell.descriptionLabel = UILabel()
+                self.syncedTabsHeaderCell.descriptionLabel?.font = DynamicFontHelper.defaultHelper.DeviceFontSmallHistoryPanel
+                self.syncedTabsHeaderCell.descriptionLabel?.textColor = UIColor.grayColor()
+                self.syncedTabsHeaderCell.descriptionLabel?.text = String(numberOfDevices) + " " + Strings.SyncedTabsTableViewCellDescription
+            }
+            self.syncedTabsHeaderCell.updateConstraints()
+        }
+    }
+}
+
+class HistoryPanelViewController: SiteTableViewController {
```

Comment:
Nit: I can see being confused between the name of this and `HistoryPanel` since they are both `UIViewController`s. Maybe rename this to `HistorySiteTableViewController` to clarify and disambiguate?